### PR TITLE
Delete note component

### PR DESCRIPTION
**Pull request by lawong-d2l**
Trello: https://trello.com/c/oGHnIeW8/30-create-delete-note-component

### CODE REVIEW

File: src/siren-action-mixin.js [hunk 1]
```diff
@@ -35,10 +35,26 @@ export const SirenActionMixin = function(superClass) {
 			}
 			var headers = new Headers();
 			this.token && headers.append('Authorization', `Bearer ${this.token}`);
-
+			var token = this.token;
 			var url = new URL(action.href, window.location.origin);
 			var body;
 
+			if(action.method === 'DELETE') {
```
Comment (jopnick) [inline src/siren-action-mixin.js]: why is this needed? shouldn't the mixin do this with the correct verb?

Comment (lawong-d2l) [reply src/siren-action-mixin.js]: @jopnick  I was trying to avoid `fields = fields || this.getSirenFields(action);`, since it has no fields in that delete action to add to a body, that isn't needed.

Comment (awikkerink) [reply src/siren-action-mixin.js]: The DELETE method definitely could have fields. I don't think this check is necessary, and adds duplicate code.

File: src/notes/note.js [hunk 2]
```diff
@@ -15,9 +15,15 @@ class Note extends LocalizationMixin(SirenActionMixin(SirenEntityMixin(PolymerEl
...
                 display: block;
             }
-        </style>
-        "[[text]]"
-        - [[date]]
+		</style>
+		<template is="dom-if" if="[[!deleted]]">
+			"[[text]]"
+			- [[date]]
+			<span on-tap="_deleteNote">Delete</span>
```
Comment (awikkerink) [inline src/notes/note.js]: Shouldn't this be a button?

Comment (lawong-d2l) [reply src/notes/note.js]: Yup!

Comment (lawong-d2l) [reply src/notes/note.js]: @awikkerink fixed

File: src/siren-action-mixin.js [hunk 3]
```diff
@@ -35,10 +35,14 @@ export const SirenActionMixin = function(superClass) {
 			}
 			var headers = new Headers();
 			this.token && headers.append('Authorization', `Bearer ${this.token}`);
-
+			var token = this.token;
 			var url = new URL(action.href, window.location.origin);
 			var body;
 
+			if (action.method === 'DELETE') {
+				return this._makeFetch(url.href, action.method, headers, token);
```
Comment (awikkerink) [inline src/siren-action-mixin.js]: I still don't understand why this is necessary. Is there a 400 error? Why do we need to avoid line 46?

Comment (lawong-d2l) [reply src/siren-action-mixin.js]: There's just no fields that come in from the `delete` action, and I'm not sending any in to the performSirenAction method. 
`Uncaught TypeError: Cannot read property 'forEach' of undefined
    at HTMLElement.getSirenFields (siren-action-mixin.js:28)
    at HTMLElement.performSirenAction (siren-action-mixin.js:45)
    at HTMLElement._deleteNote (note.js:60)
    at HTMLButtonElement.handler (template-stamp.js:90)
    at _fire (gestures.js:782)
    at Object.forward (gestures.js:1231)
    at Object.click (gestures.js:1189)
    at HTMLHtmlElement._handleNative (gestures.js:529)`
